cmd/uplink/cmd/mount.go

```diff
@@ -0,0 +1,232 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+// +build linux darwin netbsd freebsd openbsd
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"sync"
+
+	"github.com/hanwen/go-fuse/fuse"
+	"github.com/hanwen/go-fuse/fuse/nodefs"
+	"github.com/hanwen/go-fuse/fuse/pathfs"
+	"github.com/spf13/cobra"
+
+	"storj.io/storj/pkg/paths"
+	"storj.io/storj/pkg/process"
+	"storj.io/storj/pkg/ranger"
+	"storj.io/storj/pkg/storage/meta"
+	"storj.io/storj/pkg/storage/objects"
+	"storj.io/storj/pkg/utils"
+	"storj.io/storj/storage/boltdb"
+)
+
+func init() {
+	addCmd(&cobra.Command{
+		Use:   "mount",
+		Short: "Mount a bucket",
+		RunE:  mountBucket,
+	})
+}
+
+func mountBucket(cmd *cobra.Command, args []string) (err error) {
+	if len(args) == 0 {
+		return fmt.Errorf("No bucket specified for mounting")
+	}
+	if len(args) == 1 {
+		return fmt.Errorf("No destination specified")
+	}
+
+	ctx := process.Ctx(cmd)
+
+	bs, err := cfg.BucketStore(ctx)
+	if err != nil {
+		return err
+	}
+
+	u, err := utils.ParseURL(args[0])
+	if err != nil {
+		return err
+	}
+	if u.Host == "" {
+		return fmt.Errorf("No bucket specified. Please use format sj://bucket/")
+	}
+
+	store, err := bs.GetObjectStore(ctx, u.Host)
+	if err != nil {
+		return err
+	}
+
+	nfs := pathfs.NewPathNodeFs(&storjFs{FileSystem: pathfs.NewDefaultFileSystem(), ctx: ctx, store: store}, nil)
+	server, _, err := nodefs.MountRoot(args[1], nfs.Root(), nil)
+	if err != nil {
+		return fmt.Errorf("Mount failed: %v", err)
+	}
+
+	// detect control-c and unmount
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt)
+	go func() {
+		for range sig {
+			if err := server.Unmount(); err != nil {
+				fmt.Printf("Unmount failed: %v", err)
+			}
+		}
+	}()
+
+	server.Serve()
+	return nil
+}
+
+type storjFs struct {
+	ctx   context.Context
+	store objects.Store
+	pathfs.FileSystem
+}
+
+func (sf *storjFs) GetAttr(name string, context *fuse.Context) (*fuse.Attr, fuse.Status) {
+	if name == "" {
+		return &fuse.Attr{Mode: fuse.S_IFDIR | 0755}, fuse.OK
+	}
+
+	meta, err := sf.store.Meta(sf.ctx, paths.New(name))
+	if err != nil {
+		if boltdb.ErrKeyNotFound.Has(err) {
+			fmt.Println(err)
```

Comment:
Why is it printing this error and to stdout? Seems like a normal situation that could happen.